statistics/statscache_test.go

```diff
@@ -242,10 +242,65 @@ func (s *testStatsCacheSuite) TestDDL(c *C) {
 	err = h.HandleDDLEvent(<-h.DDLEventCh())
 	c.Assert(err, IsNil)
 	h.Update(is)
-	statsTbl := do.StatsHandle().GetTableStats(tableInfo)
+	statsTbl := h.GetTableStats(tableInfo)
 	c.Assert(statsTbl.Pseudo, IsFalse)
 }
 
+func (s *testStatsCacheSuite) TestVersion(c *C) {
+	store, do, err := newStoreWithBootstrap()
+	c.Assert(err, IsNil)
+	defer store.Close()
+	testKit := testkit.NewTestKit(c, store)
+	testKit.MustExec("use test")
+	testKit.MustExec("create table t1 (c1 int, c2 int)")
+	testKit.MustExec("analyze table t1")
+	is := do.InfoSchema()
+	tbl1, err := is.TableByName(model.NewCIStr("test"), model.NewCIStr("t1"))
+	c.Assert(err, IsNil)
+	tableInfo1 := tbl1.Meta()
+	h := statistics.NewHandle(testKit.Se)
+	testKit.MustExec("update mysql.stats_meta set version = 2 where table_id = ?", tableInfo1.ID)
+
+	h.Update(is)
+	statsTbl1 := h.GetTableStats(tableInfo1)
+	c.Assert(statsTbl1.Pseudo, IsFalse)
+
+	testKit.MustExec("create table t2 (c1 int, c2 int)")
+	testKit.MustExec("analyze table t2")
+	is = do.InfoSchema()
+	tbl2, err := is.TableByName(model.NewCIStr("test"), model.NewCIStr("t2"))
+	c.Assert(err, IsNil)
+	tableInfo2 := tbl2.Meta()
+	// A smaller version write, and we can still read it.
+	testKit.MustExec("update mysql.stats_meta set version = 1 where table_id = ?", tableInfo2.ID)
+	h.Update(is)
```

Comment:
Yeah, the stats will be loaded 2 times even if it was not updated, but after #3059 , The histogram will not be loaded multiple times.